src/presentational/AssignmentForm.jsx

```diff
@@ -0,0 +1,144 @@
+import { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+
+import InputElement from './InputElement';
+
+const initialState = {
+  classifications_target: '',
+  description: '',
+  duedate: '',
+  name: '',
+  students: [],
+  subjects: [],
+}
+
+
+class AssignmentForm extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.renderStudentList = this.renderStudentList.bind(this);
+    this.toggleStudent = this.toggleStudent.bind(this);
+    this.state = Object.assign({}, initialState);
+  }
+
+
+  toggleStudent(id) {
+    if (this.state.students.includes(id)) {
+      this.state.students = this.state.students.filter(student => student !== id)
+      this.setState({
+        students: this.state.students,
+      })
+    } else {
+      this.state.students.push(id);
+      this.setState({
+        students: this.state.students,
+      })
+    }
+  }
+
+  handleChange(e) {
+    let nextState = {};
```

Comment:
That's right. I used `let` because I was getting `nextState is read-only` on the next line. However I can't reproduce that now, bit confusing.